Assets/StreamingAssets/Data/Furniture.xml

```diff
@@ -277,6 +277,45 @@
     <LocalizationCode>furn_metal_smelter</LocalizationCode>
     <UnlocalizedDescription>furn_metal_smelter_desc</UnlocalizedDescription>
   </Furniture>
+	
+<Furniture objectType="Metal Furnace">
+    <Name>Metal Furnace</Name>
+    <Description>Smelts your raw ores into metal.</Description>
+    <TypeTag>Processor</TypeTag>
+    <MovementCost>1</MovementCost>
+    <PathfindingModifier>1.5</PathfindingModifier>
+    <PathfindingWeight>2</PathfindingWeight>
+    <Width>3</Width>
+    <Height>3</Height>
+
+    <BuildingJob jobTime="0" />
+
+    <JobSpotOffset X="1" Y="0" />
+
+	<FactoryInfo>
+		<PossibleProductions>
+			<ProductionChain name="Iron smelting" processingTime="4">
+				<Input>
+					<Item objectType="Raw Iron" amount="3" slotPosX="0" slotPosY="0" />
```

Comment:
sure.. but what does it mean 'inside furniture' cause you don't want to hide the input materials from the user.. it will be still placed next to machine (defined input slots)
also we are using fast little dudes to place things instead of fixed belts (like in Factorio), so having fixed input slots is ok, I think
flexible input gives you more choices and in this case prevents some setup errors, and needing to specify input slot position for every item isn't such a big extra
though it can still have default input and output slots defined for furniture.. have overrides (slotPosX and Y) as you mentioned .. 
It may be harder in have generic input slots and there may be some funky situations.. like you have two input slots, you need copper and steel.. 2 dudes will accidentally pick copper and try to feed the machine.. they will both place copper (using those 2 slots), then machine will not work.. you need to sync some job with furniture that you have to clean up one of the slots for steel..